File: Java/Implementations/Decorator/Square.java
import java.util.*;

public class Square implements Display {
	private int size;

	public Square(int size) {
		this.size = size;
	}

	public String display() {
		String hor = "-".repeat(this.size);
		String vert = "|" + " ".repeat(this.size - 2) + "|";
		StringJoiner sj = new StringJoiner("\n");
		sj.add(hor);
		for (int i = 0; i < this.size - 2; i += 1) {
			sj.add(vert);
		}
		sj.add(hor);

		return "Square of size " + this.size + "x" + this.size + "\n" + sj.toString();
	}
}

File: Haskell/Implementations/Decorator/Square.hs
module Square
    ( Square(..)
    ) where

import Data.List (genericReplicate)
import Display (Display(..))

newtype Square = MkSquare Word

instance Display Square where
    display (MkSquare size) =
        "Square of size " <> show size <> "x" <> show size <> "\n" <> square
      where
        hor      = genericReplicate size '-'
        vert     = "|" <> genericReplicate (size - 2) ' ' <> "|"
        vertSegs = genericReplicate (size - 2) vert
        square   = unlines $ [hor] <> vertSegs <> [hor]
